### PR TITLE
Add fetch vertex all props.

### PR DESCRIPTION
**Pull request by CPWstatic**
The purpose of this pr is about to help user to fetch vertex props without declaring tags.

### CODE REVIEW

**Comment by Amber1990Zhang** [comment]
Will update the doc.

**Comment by CPWstatic** [comment]
Jenkins go!

File: src/graph/test/FetchVerticesTest.cpp [hunk 1]
```diff
@@ -318,6 +318,33 @@ TEST_F(FetchVerticesTest, NonExistVertex) {
...
+        cpp2::ExecutionResponse resp;
+        auto *fmt = "FETCH PROP ON * %ld";
+        auto query = folly::stringPrintf(fmt, nonExistPlayerID);
+        auto code = client_->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+        ASSERT_EQ(nullptr, resp.get_rows());
+    }
+}
+
+TEST_F(FetchVerticesTest, FetchAll) {
```
Comment (laura-ding) [inline src/graph/test/FetchVerticesTest.cpp]: add ut in data_test which has multi-tag in vertex

Comment (CPWstatic) [reply src/graph/test/FetchVerticesTest.cpp]: Working on it.

**Comment by CPWstatic** [comment]
Jenkins go!

**Comment by CPWstatic** [comment]
Jenkins go!

File: src/graph/TraverseExecutor.cpp [hunk 2]
```diff
@@ -107,6 +107,37 @@ Status Collector::collect(VariantType &var, RowWriter *writer) {
...
+        case VAR_STR:
+            VLOG(3) << boost::get<std::string>(var);
+            (*writer) << RowWriter::ColType(nebula::cpp2::SupportedType::STRING)
+                << boost::get<std::string>(var);
+            break;
+        default:
+            std::string errMsg =
+                folly::stringPrintf("Unknown VariantType: %d", var.which());
+            LOG(ERROR) << errMsg;
+            return Status::Error(errMsg);
```
Comment (laura-ding) [inline src/graph/TraverseExecutor.cpp]: std::move

File: src/graph/test/TraverseTestBase.h [hunk 3]
```diff
@@ -768,6 +816,30 @@ AssertionResult TraverseTestBase::prepareData() {
                                << static_cast<int32_t>(code);
         }
     }
+    {
+        // Insert vertices `player'
```
Comment (laura-ding) [inline src/graph/test/TraverseTestBase.h]: modify `player` to `bachelor `

File: src/storage/query/QueryBaseProcessor.inl [hunk 4]
```diff
@@ -409,6 +409,47 @@ kvstore::ResultCode QueryBaseProcessor<REQ, RESP>::collectVertexProps(
...
+            continue;
+        }
+        missedKey = false;
+        auto val = iter->val().toString();
+        auto ver = RowReader::getSchemaVer(val);
+        if (ver < 0) {
+            LOG(ERROR) << "Found schema version negative " << ver;
+            continue;
+        }
+        auto tagId = NebulaKeyUtils::getTagId(iter->key());
```
Comment (laura-ding) [inline src/storage/query/QueryBaseProcessor.inl]: function `key()` and `val()` returns the new string, and here call `key()` and `val()` multi times, you can define local var to save them to reduce string copy.

Comment (CPWstatic) [reply src/storage/query/QueryBaseProcessor.inl]: We actually need to copy them.

File: src/storage/query/QueryBoundProcessor.cpp [hunk 5]
```diff
@@ -86,6 +86,14 @@ kvstore::ResultCode QueryBoundProcessor::processVertex(PartitionID partId, Verte
             }
         }
         vResp.set_tag_data(std::move(td));
+    } else if (tagContexts_.empty() && onlyVertexProps_) {
```
Comment (dangleptr) [inline src/storage/query/QueryBoundProcessor.cpp]: Implement the "collect all props for one vertex"  logic inside QueryVertexPropsProcessor

File: src/storage/query/QueryBaseProcessor.inl [hunk 6]
```diff
@@ -409,6 +409,47 @@ kvstore::ResultCode QueryBaseProcessor<REQ, RESP>::collectVertexProps(
...
+        return ret;
+    }
+
+    bool missedKey = true;
+    for (; iter && iter->valid(); iter->next()) {
+        if (!NebulaKeyUtils::isVertex(iter->key())) {
+            continue;
+        }
+        missedKey = false;
+        auto val = iter->val().toString();
```
Comment (dangleptr) [inline src/storage/query/QueryBaseProcessor.inl]: We don't need to call  "toString" here.

File: src/graph/FetchVerticesExecutor.cpp [hunk 7]
```diff
@@ -312,5 +322,69 @@ Status FetchVerticesExecutor::setupVidsFromRef() {
     vids_ = std::move(result).value();
     return Status::OK();
 }
+
+void FetchVerticesExecutor::processAllPropsResult(RpcResponse &&result) {
```
Comment (Shylock-Hg) [inline src/graph/FetchVerticesExecutor.cpp]: Could combine with `processResult`?

File: src/storage/query/QueryBaseProcessor.inl [hunk 8]
```diff
@@ -409,6 +409,49 @@ kvstore::ResultCode QueryBaseProcessor<REQ, RESP>::collectVertexProps(
     }
     return ret;
 }
+template<typename REQ, typename RESP>
+kvstore::ResultCode QueryBaseProcessor<REQ, RESP>::collectVertexProps(
```
Comment (dangleptr) [inline src/storage/query/QueryBaseProcessor.inl]: Move this function into QueryVertexPropsProcessor

**Comment by jude-zhu** [comment]
close #1183

File: src/graph/FetchVerticesExecutor.cpp (hunk 7 above)
Comment (CPWstatic) [reply src/graph/FetchVerticesExecutor.cpp]: Better to separate them, i think.

File: src/graph/test/TraverseTestBase.h [hunk 9]
```diff
@@ -768,6 +816,30 @@ AssertionResult TraverseTestBase::prepareData() {
...
+            query += bachelor.name();
+            query += "\"";
+            query += ",";
+            query += bachelor.major();
+            query += "),\n\t";
+        }
+        query.resize(query.size() - 3);
+        auto code = client_->execute(query, resp);
+        if (code != cpp2::ErrorCode::SUCCEEDED) {
+            return TestError() << "Insert `players' failed: "
```
Comment (laura-ding) [inline src/graph/test/TraverseTestBase.h]: modify `players` to `bachelor `

File: src/storage/query/QueryVertexPropsProcessor.cpp [hunk 10]
```diff
@@ -14,18 +14,100 @@ namespace nebula {
...
+            LOG(ERROR) << "Found schema version negative " << ver;
+            continue;
+        }
+        auto tagId = NebulaKeyUtils::getTagId(key);
+        auto result = tagIds.emplace(tagId);
+        if (!result.second) {
+            // Already found the latest version.
+            continue;
+        }
+        VLOG(3) << "Found tag " << tagId << " for vId" << vId;
```
Comment (laura-ding) [inline src/storage/query/QueryVertexPropsProcessor.cpp]: add space after `vId`

**Pull request merged**